### ISSUE TITLE
[bug] Error installing conan on Big Sur running on Apple Silicon

### ISSUE THREAD

**Issue by acronce** [conan-io/conan#7286]
### Environment Details (include every applicable attribute)
  * Operating System+version: macOS Big Sur 11.0 Beta (20A5299w) on Apple A12Z Bionic
  * Compiler+version: Apple clang version 12.0.0 (clang-1200.0.22.19)
  * Conan version: 1.26.1
  * Python version: 3.8.2

### Steps to reproduce (Include if Applicable)
sudo pip3 install conan

### Logs (Executed commands with output) (Include/Attach if Applicable)
The problem is a compile error building "cryptography". I realize that this might be a Python issue with a conan requirement and not directly related to the conan project. But it is keeping me from building any of our projects that use conan on Apple Silicon systems.

If you feel this should reported to another project please let me know.

In the meantime, here's the build log (search below for 'error:' without quotes):

```
  Building wheel for cryptography (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /Applications/Xcode-beta.app/Contents/Developer/usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/tmp/pip-install-kkmw5tcb/cryptography/setup.py'"'"'; __file__='"'"'/private/tmp/pip-install-kkmw5tcb/cryptography/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/tmp/pip-wheel-jnxw961w --python-tag cp38
       cwd: /private/tmp/pip-install-kkmw5tcb/cryptography/
  Complete output (150 lines):
  c/_cffi_backend.c:6245:9: warning: 'ffi_prep_closure' is deprecated [-Wdeprecated-declarations]
      if (ffi_prep_closure(closure, &cif_descr->cif,
          ^
  /Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/ffi/ffi.h:350:18: note: 'ffi_prep_closure' has been explicitly marked deprecated here
    __attribute__((deprecated))
                   ^
  1 warning generated.
  c/_cffi_backend.c:6245:9: error: implicit declaration of function 'ffi_prep_closure' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
      if (ffi_prep_closure(closure, &cif_descr->cif,
          ^
  1 error generated.
  Traceback (most recent call last):
    File "/Applications/Xcode-beta.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/distutils/unixccompiler.py", line 117, in _compile
      self.spawn(compiler_so + cc_args + [src, '-o', obj] +
    File "/Applications/Xcode-beta.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/distutils/ccompiler.py", line 910, in spawn
      spawn(cmd, dry_run=self.dry_run)
    File "/Applications/Xcode-beta.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/distutils/spawn.py", line 36, in spawn
      _spawn_posix(cmd, search_path, dry_run=dry_run)
    File "/Applications/Xcode-beta.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/distutils/spawn.py", line 157, in _spawn_posix
      raise DistutilsExecError(
  distutils.errors.DistutilsExecError: command 'xcrun' failed with exit status 1
```

**Comment by memsharded** [comment]
Well, if you can't ``sudo pip3 install cryptography`` in that platform, I guess, yes, you should be probably reporting this in https://github.com/pyca/cryptography first (please link the issue here, so we can track, if you do, thanks!)

**Comment by acronce** [comment]
Thanks for the fast reply! Yes I see the same error when just trying to install cryptography. 

Here's the link to the issue over there: https://github.com/pyca/cryptography/issues/5291

**Comment by memsharded** [comment]
Cool, thanks for reporting this! I see they labeled it as ``blocked-on-upstream``, which means that they also have transitive dependencies to cffi that is failing for them.

I would expect the whole python ecosystem to struggle for a while to provide support for Apple Silicon, so I would suggest to start trying running Conan with some emulation mode or something meanwhile.

**Comment by acronce** [comment]
I suppose I could try manually installing cryptography from another (Intel) installation. I've run several Intel apps under Rosetta 2 (including CMake) and it just seems to work. Kind of amazing, really.

**Comment by acronce** [comment]
With some help and was able to build the necessary prerequisites and install Conan. The last issue was the need to change my settings.yml to include Apple Clang 12:

```
    apple-clang: &apple_clang
        version: ["5.0", "5.1", "6.0", "6.1", "7.0", "7.3", "8.0", "8.1", "9.0", "9.1", "10.0", "11.0", "12.0"]
```

Probably a good idea to change the Conan code that creates settings.yml to do this so that new installations on Apple Silicon machines will work with Xcode 12 out of the box. 

Let me know if you want me to create a separate case for that.

Next interesting task will be to figure out how to get Universal binaries working with CMake and Conan. Universal (i.e. Intel and ARM) binaries will be important for at least the next 5 years.

**Comment by memsharded** [comment]
Hi @acronce 

Yes, definitely version 12.0 can be added to the default settings.yml for next Conan 1.28 release. You can open a new issue, or if you want to directly contribute a PR to add it, that would be great, shouldn't be very difficult, please do. Please take into account that if the default settings.yml change, it is necessary to change the last line in ``migration_settings.py`` to store the same settings.yml value for ``settings_1_28_0``.

**Comment by acronce** [comment]
Sounds straightforward enough. I'll do a PR this week.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/7722 has added apple-clang 12, released in Conan 1.29.2

**Comment by memsharded** [comment]
To follow up (cc/ @rconde01) : https://github.com/conan-io/conan/pull/8075 is going to remove ``cryptography`` in OSX as a requirement, apparently it is not needed anymore. If things work without it, then we can consider this solved and close it.

Please upgrade to Conan 1.32 when it is released and report if it doesn't work. Thanks!
